Conceptpower+Spring/src/main/java/edu/asu/conceptpower/servlet/core/impl/IndexService.java

```diff
@@ -0,0 +1,108 @@
+import org.springframework.stereotype.Service;
+
+import edu.asu.conceptpower.servlet.core.ConceptEntry;
+import edu.asu.conceptpower.servlet.core.ErrorConstants;
+import edu.asu.conceptpower.servlet.core.IIndexService;
+import edu.asu.conceptpower.servlet.exceptions.IndexerRunningException;
+import edu.asu.conceptpower.servlet.exceptions.LuceneException;
+import edu.asu.conceptpower.servlet.lucene.impl.LuceneUtility;
+
+/**
+ * This class acts as a single point of access to lucene. This class performs
+ * searching, insertion, deletion and indexing in lucene
+ * 
+ * @author karthikeyanmohan
+ *
+ */
+@Service
+public class IndexService implements IIndexService {
+
+	@Autowired
+	private LuceneUtility luceneUtility;
+
+	/**
+	 * This field makes sure indexer runs only once when two different admins
+	 * gives index command at same time
+	 */
+	private AtomicBoolean indexerRunningFlag = new AtomicBoolean(false);
+
+	/**
+	 * This method searches in lucene on a particular condition fed through
+	 * fieldMap
+	 */
+	@Override
+	public ConceptEntry[] searchForConcepts(Map<String, String> fieldMap, String operator)
+			throws LuceneException, IllegalAccessException, IndexerRunningException {
+
+		if (indexerRunningFlag.get()) {
+			throw new IndexerRunningException(ErrorConstants.INDEXER_RUNNING);
+		}
+
+		return luceneUtility.queryIndex(fieldMap, operator);
+	}
+
+	/**
+	 * This method inserts concepts into lucene
+	 * 
+	 * @throws IndexerRunningException
+	 */
+	@Override
+	public void insertConcept(ConceptEntry entry)
+			throws IllegalAccessException, LuceneException, IndexerRunningException {
+		if (indexerRunningFlag.get()) {
+			throw new IndexerRunningException(ErrorConstants.INDEXER_RUNNING);
+
+		}
+		luceneUtility.insertConcept(entry);
+	}
+
+	/**
+	 * This method deletes the concept in lucene index based on id of the
+	 * concept
+	 */
+	@Override
+	public void deleteById(String id) throws LuceneException, IndexerRunningException {
+		if (indexerRunningFlag.get()) {
+			throw new IndexerRunningException(ErrorConstants.INDEXER_RUNNING);
+		}
+		luceneUtility.deleteById(id);
+
+	}
+
+	/**
+	 * This method deletes the index in lucene
+	 */
+	@Override
+	public void deleteIndexes() throws LuceneException, IndexerRunningException {
+		if (!indexerRunningFlag.compareAndSet(false, true)) {
+			throw new IndexerRunningException(ErrorConstants.INDEXER_RUNNING);
+		}
+		luceneUtility.deleteIndexes();
+		indexerRunningFlag.set(false);
+	}
+
+	/**
+	 * This method indexes concepts in lucene and runs indexer only once
+	 */
+	@Override
+	public void indexConcepts()
+			throws LuceneException, IllegalArgumentException, IllegalAccessException, IndexerRunningException {
+		if (!indexerRunningFlag.compareAndSet(false, true)) {
+			throw new IndexerRunningException(ErrorConstants.INDEXER_RUNNING);
+		}
+		luceneUtility.indexConcepts();
+		indexerRunningFlag.set(false);
+	}
+
+	@Override
+	public boolean checkIndexerStatus() {
```

Comment:
bad method name. Especially with boolean return values, the name should indicate how return value and method name relate, e.g. isWhatEverYouDoHappening -> yes/no -> true/false
in this case, e.g.: isIndexerRunning